### PR TITLE
ICMSLST-2631 - Active User Report

### PR DESCRIPTION
**Pull request by marcuspp**

### CODE REVIEW

File: web/reports/interfaces.py [hunk 1]
```diff
@@ -1174,10 +1176,19 @@ def get_queryset(self) -> QuerySet:
...
         )
+        _permissions_filter = Q(
+            Q(groups__isnull=True)
+            | Q(
+                groups__name__in=[
+                    Perms.obj.importer.get_group_name(),
+                    Perms.obj.exporter.get_group_name(),
+                ]
+            )
+        )
```
[the remark this answers is not in the document]
Comment (MattHolmes123) [reply web/reports/interfaces.py:1187]: We also have a `get_users_with_permission` function in `work/icms/web/permissions/service.py` that is used in several other places.

File: web/reports/interfaces.py [hunk 2]
```diff
@@ -1155,3 +1161,66 @@ def extra_fields(self, ia: ImportApplication) -> dict[str, Any]:
...
+
+    def get_queryset(self) -> QuerySet:
+        _filter = Q(
+            date_joined__date__range=(
+                self.filters.date_from,
+                self.filters.date_to,
+            )
+        )
+        _permissions_filter = Q(
+            Q(groups__isnull=True)
```
Comment (MattHolmes123) [inline web/reports/interfaces.py:1180]: Why do we need to return users with no groups?

File: web/reports/interfaces.py [hunk 3]
```diff
@@ -1155,3 +1161,66 @@ def extra_fields(self, ia: ImportApplication) -> dict[str, Any]:
...
+                groups__permissions__codename__in=[
+                    Perms.sys.exporter_access.codename,
+                    Perms.sys.importer_access.codename,
+                ]
+            )
+        )
+        return (
+            user_list_view_qs()
+            .filter(is_active=True)
+            .filter(_filter, _permissions_filter)
```
Comment (MattHolmes123) [inline web/reports/interfaces.py:1191]: `.filter(_filter, _permissions_filter, is_active=True)`

File: web/reports/interfaces.py [hunk 4]
```diff
@@ -1155,3 +1161,66 @@ def extra_fields(self, ia: ImportApplication) -> dict[str, Any]:
...
+                exporters=ArraySubquery(
+                    ExporterUserObjectPermission.objects.filter(user__pk=OuterRef("pk"))
+                    .values_list("content_object__name", flat=True)
+                    .distinct()
+                ),
+                importers=ArraySubquery(
+                    ImporterUserObjectPermission.objects.filter(user__pk=OuterRef("pk"))
+                    .values_list("content_object__name", flat=True)
+                    .distinct()
+                ),
```
Comment (MattHolmes123) [inline web/reports/interfaces.py:1202]: This returns the parent company for agents, is that the desired behaviour?

**Review by MattHolmes123 — APPROVED** [review]

File: web/reports/interfaces.py (hunk 2 above)
Comment (marcuspp) [reply web/reports/interfaces.py:1180]: Users who have logged in and haven't created an access request need to be included

Comment (marcuspp) [reply web/reports/interfaces.py:1180]: The filtering is really exclude caseworkers/staff and include everyone else

**Pull request merged**